arch/risc-v/src/esp32c3/esp32c3_irq.c

```diff
@@ -355,7 +355,7 @@ void esp32c3_free_cpuint(uint8_t periphid)
 IRAM_ATTR uintptr_t *esp32c3_dispatch_irq(uintptr_t mcause, uintptr_t *regs)
 {
   int irq;
-  uintptr_t *mepc = regs;
+  uint8_t cpuint = mcause & RISCV_IRQ_MASK;
```

Comment:
is `uint8_t` a sufficient holder for result? `RISCV_IRQ_MASK` seems to be a `uintptr_t` size.

Reply:
The result should be between 0-31 (number of CPU interrupts).

Reply:
Yes. And using `uintptr_t` + `DEBUGASSERT(cpuint <= ESP32C3_CPUINT_MAX);` seems to provide better catching mechanism than `uint8_t` because upper bits are not cutoff